pkg/network/proxyimpl/hybrid/proxy.go

```diff
@@ -143,6 +148,21 @@ func (p *HybridProxier) OnServiceUpdate(oldService, service *corev1.Service) {
 	}
 }
 
+// cleanupState handles the deletion of endpoints and service in any order.
+// svcName should be the NamespacedName of a service (or endpoint).
+func (p *HybridProxier) cleanupState(svcName types.NamespacedName) {
+	_, isPendingDeletion := p.pendingDeletion[svcName]
```

Comment:
Does this map need to be locked? IIRC OnServiceDelete() and OnEndpointsDelete() can be called at the same time (because they are called from different goroutines) and that could result in concurrent map access.

Reply:
I remember discussing this with Casey, he said it wasn't necessary (https://bugzilla.redhat.com/show_bug.cgi?id=1857743#c14)

Reply:
so if I remember correctly what he had mentioned was that since both `OnServiceDelete` and `OnEndpointsDelete` use `p.usingUserspaceLock.Lock()` we are safe while using this map.

Reply:
@dcbw : ^ (PTAL when you have time, patch manager won't approve otherwise.)

Reply:
Ok, thanks for the explanation.